sys/riscv/riscv/pmap.c

```diff
@@ -1757,8 +1757,8 @@ pmap_alloc_l2(pmap_t pmap, vm_offset_t va, struct rwlock **lockp)
 		l2pg->ref_count++;
 	} else {
 		/* Allocate a L2 page. */
-		l2pindex = pmap_l2_pindex(va) >> Ln_ENTRIES_SHIFT;
-		l2pg = _pmap_alloc_l3(pmap, NUL2E + l2pindex, lockp);
+		ptdpindex = pmap_l1_pindex(va);
+		l2pg = _pmap_alloc_l3(pmap, ptdpindex, lockp);
```

Comment:
Oh by the way, I think there might be more opportunities to use these macros; see `_pmap_alloc_l3()`.

Reply:
I have checked that functions that are calling _pmap_alloc_l3 are all using pmap_l1/l2_pindex to get the pagetable page index for _pmap_alloc_l3. On the other hand, _pmap_alloc_l3 does not need pmap_l1/l2_pindex. When it recusively calls itself, it needs to deduce the pindex for higher level pagetable page by itself.